{
  "SchemaVersion": 2,
  "ArtifactName": "jsons/docker_scout/sboms/aerospike:ee-7.0.0.4_1_sbom.json",
  "ArtifactType": "spdx",
  "Metadata": {
    "OS": {
      "Family": "none",
      "Name": ""
    },
    "ImageConfig": {
      "architecture": "",
      "created": "0001-01-01T00:00:00Z",
      "os": "",
      "rootfs": {
        "type": "",
        "diff_ids": null
      },
      "config": {}
    }
  },
  "Results": [
    {
      "Target": "Python",
      "Class": "lang-pkgs",
      "Type": "python-pkg",
      "Vulnerabilities": [
        {
          "VulnerabilityID": "CVE-2023-50782",
          "PkgName": "cryptography",
          "InstalledVersion": "41.0.7",
          "FixedVersion": "42.0.0",
          "Status": "fixed",
          "Layer": {},
          "SeveritySource": "ghsa",
          "PrimaryURL": "https://avd.aquasec.com/nvd/cve-2023-50782",
          "PkgRef": "pkg:pypi/cryptography@41.0.7",
          "DataSource": {
            "ID": "ghsa",
            "Name": "GitHub Security Advisory pip",
            "URL": "https://github.com/advisories?query=type%3Areviewed+ecosystem%3Apip"
          },
          "Title": "python-cryptography: Bleichenbacher timing oracle attack against RSA decryption - incomplete fix for CVE-2020-25659",
          "Description": "A flaw was found in the python-cryptography package. This issue may allow a remote attacker to decrypt captured messages in TLS servers that use RSA key exchanges, which may lead to exposure of confidential or sensitive data.",
          "Severity": "HIGH",
          "CweIDs": [
            "CWE-203",
            "CWE-208"
          ],
          "CVSS": {
            "ghsa": {
              "V3Vector": "CVSS:3.1/AV:N/AC:L/PR:N/UI:N/S:U/C:H/I:N/A:N",
              "V3Score": 7.5
            },
            "nvd": {
              "V3Vector": "CVSS:3.1/AV:N/AC:L/PR:N/UI:N/S:U/C:H/I:N/A:N",
              "V3Score": 7.5
            },
            "redhat": {
              "V3Vector": "CVSS:3.1/AV:N/AC:L/PR:N/UI:N/S:U/C:H/I:N/A:N",
              "V3Score": 7.5
            }
          },
          "References": [
            "https://access.redhat.com/security/cve/CVE-2023-50782",
            "https://bugzilla.redhat.com/show_bug.cgi?id=2254432",
            "https://github.com/openssl/openssl/pull/13817",
            "https://github.com/pyca/cryptography",
            "https://github.com/pyca/cryptography/issues/9785",
            "https://nvd.nist.gov/vuln/detail/CVE-2023-50782",
            "https://people.redhat.com/~hkario/marvin/",
            "https://ubuntu.com/security/notices/USN-6663-1",
            "https://ubuntu.com/security/notices/USN-6673-1",
            "https://ubuntu.com/security/notices/USN-6673-2",
            "https://www.cve.org/CVERecord?id=CVE-2023-50782"
          ],
          "PublishedDate": "2024-02-05T21:15:11.183Z",
          "LastModifiedDate": "2024-02-26T16:27:48.08Z"
        },
        {
          "VulnerabilityID": "CVE-2024-26130",
          "PkgName": "cryptography",
          "InstalledVersion": "41.0.7",
          "FixedVersion": "42.0.4",
          "Status": "fixed",
          "Layer": {},
          "SeveritySource": "ghsa",
          "PrimaryURL": "https://avd.aquasec.com/nvd/cve-2024-26130",
          "PkgRef": "pkg:pypi/cryptography@41.0.7",
          "DataSource": {
            "ID": "ghsa",
            "Name": "GitHub Security Advisory pip",
            "URL": "https://github.com/advisories?query=type%3Areviewed+ecosystem%3Apip"
          },
          "Title": "python-cryptography: NULL pointer dereference with pkcs12.serialize_key_and_certificates when called with a non-matching certificate and private key and an hmac_hash override",
          "Description": "cryptography is a package designed to expose cryptographic primitives and recipes to Python developers. Starting in version 38.0.0 and prior to version 42.0.4, if `pkcs12.serialize_key_and_certificates` is called with both a certificate whose public key did not match the provided private key and an `encryption_algorithm` with `hmac_hash` set (via `PrivateFormat.PKCS12.encryption_builder().hmac_hash(...)`, then a NULL pointer dereference would occur, crashing the Python process. This has been resolved in version 42.0.4, the first version in which a `ValueError` is properly raised.",
          "Severity": "HIGH",
          "CweIDs": [
            "CWE-476"
          ],
          "CVSS": {
            "ghsa": {
              "V3Vector": "CVSS:3.1/AV:N/AC:L/PR:N/UI:N/S:U/C:N/I:N/A:H",
              "V3Score": 7.5
            },
            "redhat": {
              "V3Vector": "CVSS:3.1/AV:N/AC:L/PR:N/UI:N/S:U/C:N/I:N/A:H",
              "V3Score": 7.5
            }
          },
          "References": [
            "https://access.redhat.com/security/cve/CVE-2024-26130",
            "https://github.com/pyca/cryptography",
            "https://github.com/pyca/cryptography/commit/97d231672763cdb5959a3b191e692a362f1b9e55",
            "https://github.com/pyca/cryptography/pull/10423",
            "https://github.com/pyca/cryptography/security/advisories/GHSA-6vqw-3v5j-54x4",
            "https://nvd.nist.gov/vuln/detail/CVE-2024-26130",
            "https://ubuntu.com/security/notices/USN-6673-1",
            "https://www.cve.org/CVERecord?id=CVE-2024-26130"
          ],
          "PublishedDate": "2024-02-21T17:15:09.863Z",
          "LastModifiedDate": "2024-02-22T19:07:27.197Z"
        },
        {
          "VulnerabilityID": "CVE-2024-0727",
          "PkgName": "cryptography",
          "InstalledVersion": "41.0.7",
          "FixedVersion": "42.0.2",
          "Status": "fixed",
          "Layer": {},
          "SeveritySource": "ghsa",
          "PrimaryURL": "https://avd.aquasec.com/nvd/cve-2024-0727",
          "PkgRef": "pkg:pypi/cryptography@41.0.7",
          "DataSource": {
            "ID": "ghsa",
            "Name": "GitHub Security Advisory pip",
            "URL": "https://github.com/advisories?query=type%3Areviewed+ecosystem%3Apip"
          },
          "Title": "openssl: denial of service via null dereference",
          "Description": "Issue summary: Processing a maliciously formatted PKCS12 file may lead OpenSSL\nto crash leading to a potential Denial of Service attack\n\nImpact summary: Applications loading files in the PKCS12 format from untrusted\nsources might terminate abruptly.\n\nA file in PKCS12 format can contain certificates and keys and may come from an\nuntrusted source. The PKCS12 specification allows certain fields to be NULL, but\nOpenSSL does not correctly check for this case. This can lead to a NULL pointer\ndereference that results in OpenSSL crashing. If an application processes PKCS12\nfiles from an untrusted source using the OpenSSL APIs then that application will\nbe vulnerable to this issue.\n\nOpenSSL APIs that are vulnerable to this are: PKCS12_parse(),\nPKCS12_unpack_p7data(), PKCS12_unpack_p7encdata(), PKCS12_unpack_authsafes()\nand PKCS12_newpass().\n\nWe have also fixed a similar issue in SMIME_write_PKCS7(). However since this\nfunction is related to writing data we do not consider it security significant.\n\nThe FIPS modules in 3.2, 3.1 and 3.0 are not affected by this issue.",
          "Severity": "MEDIUM",
          "CVSS": {
            "ghsa": {
              "V3Vector": "CVSS:3.1/AV:L/AC:L/PR:N/UI:R/S:U/C:N/I:N/A:H",
              "V3Score": 5.5
            },
            "nvd": {
              "V3Vector": "CVSS:3.1/AV:L/AC:L/PR:N/UI:R/S:U/C:N/I:N/A:H",
              "V3Score": 5.5
            },
            "redhat": {
              "V3Vector": "CVSS:3.1/AV:L/AC:L/PR:N/UI:R/S:U/C:N/I:N/A:H",
              "V3Score": 5.5
            }
          },
          "References": [
            "https://access.redhat.com/security/cve/CVE-2024-0727",
            "https://github.com/alexcrichton/openssl-src-rs/commit/add20f73b6b42be7451af2e1044d4e0e778992b2",
            "https://github.com/github/advisory-database/pull/3472",
            "https://github.com/openssl/openssl/commit/09df4395b5071217b76dc7d3d2e630eb8c5a79c2",
            "https://github.com/openssl/openssl/commit/775acfdbd0c6af9ac855f34969cdab0c0c90844a",
            "https://github.com/openssl/openssl/commit/d135eeab8a5dbf72b3da5240bab9ddb7678dbd2c",
            "https://github.com/openssl/openssl/pull/23362",
            "https://github.com/pyca/cryptography/commit/3519591d255d4506fbcd0d04037d45271903c64d",
            "https://github.openssl.org/openssl/extended-releases/commit/03b3941d60c4bce58fab69a0c22377ab439bc0e8",
            "https://github.openssl.org/openssl/extended-releases/commit/aebaa5883e31122b404e450732dc833dc9dee539",
            "https://nvd.nist.gov/vuln/detail/CVE-2024-0727",
            "https://security.netapp.com/advisory/ntap-20240208-0006",
            "https://security.netapp.com/advisory/ntap-20240208-0006/",
            "https://ubuntu.com/security/notices/USN-6622-1",
            "https://ubuntu.com/security/notices/USN-6632-1",
            "https://ubuntu.com/security/notices/USN-6709-1",
            "https://www.cve.org/CVERecord?id=CVE-2024-0727",
            "https://www.openssl.org/news/secadv/20240125.txt"
          ],
          "PublishedDate": "2024-01-26T09:15:07.637Z",
          "LastModifiedDate": "2024-02-08T10:15:13.91Z"
        }
      ]
    }
  ]
}
